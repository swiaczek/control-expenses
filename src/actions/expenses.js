import { v4 as uuid } from "uuid";

//Add expense function

export const addExpense = ({
  description = "",
  note = "",
  amount = 0,
  createdAt = 0,
} = {}) => ({
  type: "ADD_EXPENSE",
  expense: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt,
  },
});
//Remove Expense Function
export const removeExpense = (id) => ({
  type: "REMOVE_EXPENSE",
  id,
});

//Edit Expense Function
export const editExpense = (id, updates) => ({
  type: "EDIT_EXPENSE",
  id,
  updates,
});
