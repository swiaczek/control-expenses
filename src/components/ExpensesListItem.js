import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpensesListItem = ({ id, description, amount, createdAt }) => {
  return (
    <Link className="list-item" to={`/edit/${id}`}>
      <div>
        <h4 className="list-item__title">{description}</h4>
        <span className="list-item__subtitle">
          {numeral(amount / 100).format("0,0.0")}â‚¬
        </span>
      </div>
      <span>{moment(createdAt).format("Do MMM, YYYY")}</span>
    </Link>
  );
};

export default ExpensesListItem;
