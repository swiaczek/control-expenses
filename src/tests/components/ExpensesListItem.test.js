import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";
import ExpensesListItem from "../../components/ExpensesListItem";

test("should render ExpenseListItem with fixture data", () => {
  const wrapper = shallow(<ExpensesListItem {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
