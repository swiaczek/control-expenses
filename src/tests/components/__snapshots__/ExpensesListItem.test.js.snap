// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with fixture data 1`] = `
<Link
  className="list-item"
  to="/edit/1"
>
  <div>
    <h4
      className="list-item__title"
    >
      Bike
    </h4>
    <span
      className="list-item__subtitle"
    >
      0.1
      â‚¬
    </span>
  </div>
  <span>
    1st Jan, 1970
  </span>
</Link>
`;
